staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -49,11 +62,195 @@ func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object)
 			})
 		}
 
-		pod.Spec.HostIPC = true
-		pod.Spec.HostNetwork = true
-		pod.Spec.HostPID = true
+		setHostNamespace(pod, true)
 		return nil
 	default:
 		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
 	}
 }
+
+func isPodCopy(pod *corev1.Pod, target runtime.Object) bool {
+	if asserted, ok := target.(*corev1.Pod); !ok {
+		return false
+	} else {
+		return pod != asserted
+	}
+}
+
+type debugStyle int
+
+const (
+	styleEphemeral debugStyle = iota
```

Comment:
I think the consistent prefix makes it clearer at call sites. As another example, all the profile name strings are prefixed with the sub-string "profile".

Because all the profiles have to do this exact branching logic based on the debug technique, I think it's good to have this to slightly reduce boilerplate code, improve readability, and improve consistency in profile's implementations.

Reply:
I'd suggest;

```go
Ephemeral debugType = iot
```

Reply:
Dropped in 98878e9a835d52da57cc439907a25b560117446b